### PR TITLE
Add comments to events based cache code

### PR DESCRIPTION
**Pull request by faisal-memon**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [ ] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**
Events based cache

**Description of change**
- Adds some comments to make the workflow more clear.
- Update a pair of if statements to make the code easier to read.

### CODE REVIEW

File: pkg/server/endpoints/authorized_entryfetcher_attested_nodes.go [hunk 1]
```diff
@@ -111,8 +113,8 @@ func (a *attestedNodes) updateCache(ctx context.Context) error {
 		// were skipped over and need to be queued in case they show up later.
 		// This can happen when a long running transaction allocates an event ID but a shorter transaction
 		// comes in after, allocates and commits the ID first. If a read comes in at this moment, the event id for
-		// the longer running transaction will be skipped over
-		if !a.firstEventTime.IsZero() && event.EventID != a.lastEventID+1 {
+		// the longer running transaction will be skipped over.
+		if !a.firstEventTime.IsZero() {
```
Comment (faisal-memon) [inline pkg/server/endpoints/authorized_entryfetcher_attested_nodes.go:117]: Following same style suggested by this [code review comment](https://github.com/spiffe/spire/pull/5289#discussion_r1683227127).

File: pkg/server/endpoints/authorized_entryfetcher_registration_entries.go [hunk 2]
```diff
@@ -124,8 +126,8 @@ func (a *registrationEntries) updateCache(ctx context.Context) error {
 		// were skipped over and need to be queued in case they show up later.
 		// This can happen when a long running transaction allocates an event ID but a shorter transaction
 		// comes in after, allocates and commits the ID first. If a read comes in at this moment, the event id for
-		// the longer running transaction will be skipped over
-		if !a.firstEventTime.IsZero() && event.EventID != a.lastEventID+1 {
+		// the longer running transaction will be skipped over.
+		if !a.firstEventTime.IsZero() {
```
Comment (faisal-memon) [inline pkg/server/endpoints/authorized_entryfetcher_registration_entries.go:130]: Following same style suggested by this [code review comment](https://github.com/spiffe/spire/pull/5289#discussion_r1683227127).

File: pkg/server/endpoints/authorized_entryfetcher_attested_nodes.go [hunk 3]
```diff
@@ -188,7 +188,6 @@ func (a *attestedNodes) replayMissedEvents(ctx context.Context) {
 		switch status.Code(err) {
 		case codes.OK:
 		case codes.NotFound:
-			log.Debug("Event not yet populated in database")
```
[the remark this answers is not in the document]
Comment (faisal-memon) [reply pkg/server/endpoints/authorized_entryfetcher_attested_nodes.go:185]: Thanks @stevend-uber for the feedback. Ill work on making the metrics changes. I didnt understand the side question though.

[the remark this answers is not in the document]
Comment (stevend-uber) [reply pkg/server/endpoints/authorized_entryfetcher_attested_nodes.go:185]: For #1: Mostly why we don't do anything for `case codes.NotFound:`. 

Upon initial reading of the code without your or my context, another contributor is likely to add a log here and we're back where we started :) 

It's not blocking so I'm cool just adding it later :)

**Review by stevend-uber — APPROVED** [review]

File: pkg/server/endpoints/authorized_entryfetcher_attested_nodes.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (edwbuck) [reply pkg/server/endpoints/authorized_entryfetcher_attested_nodes.go:185]: The logging for not found is likely to be better handled in #5341 as part of the backoff algorithm, and we intend to add a guage to track the entries in #4720.  With these items, which are more specific, I'll ask that we not attempt non-comment changes in this PR, as it makes the PR lose focus and then velocity reduces.

If the two PRs mentioned above don't handle the issue, please search #5327, #5342, #4720, or #5349 to see if it is covered there, and if not, open a new PR.

**Comment by edwbuck** [comment]
@azdagron Please see if you can provide a second review on this item, I would review it, but I want eyes further from the solution to see if the comments read better to a wider audience.

File: pkg/server/endpoints/authorized_entryfetcher_attested_nodes.go [hunk 4]
```diff
@@ -143,6 +145,10 @@ func (a *attestedNodes) updateCache(ctx context.Context) error {
 	return nil
 }
 
+// missedStartupEvents will check for any events come in with an ID less than the first event ID we receive.
+// For example if the first event ID we receive is 3, this function will check for any IDs less than that.
+// If event ID 2 comes in later on, due to a long running transaction, this function will update the cache
+// with the information from this event. This function will run the first sqlTransactionTimeout hours after startup.
```
Comment (azdagron) [inline pkg/server/endpoints/authorized_entryfetcher_attested_nodes.go:151]: nit: sqlTransactionTimeout is not a unit of hours
```suggestion
// with the information from this event. This function will run until time equal to sqlTransactionTimeout has elapsed after startup.
```

File: pkg/server/endpoints/authorized_entryfetcher_registration_entries.go [hunk 5]
```diff
@@ -31,7 +31,8 @@ type registrationEntries struct {
 	sqlTransactionTimeout   time.Duration
 }
 
-// buildRegistrationEntriesCache Fetches all registration entries and adds them to the cache
+// buildRegistrationEntriesCache Fetches all registration entries and adds them to the cache.
```
Comment (azdagron) [inline pkg/server/endpoints/authorized_entryfetcher_registration_entries.go:34]:
```suggestion
// buildRegistrationEntriesCache fetches all registration entries and adds them to the cache.
```

File: pkg/server/endpoints/authorized_entryfetcher_registration_entries.go [hunk 6]
```diff
@@ -49,6 +50,8 @@ func buildRegistrationEntriesCache(ctx context.Context, log logrus.FieldLogger,
 			firstEventID = event.EventID
 			firstEventTime = now
 		} else {
+			// After getting the first event, search for any gaps in the event stream, from the first event to the last event.
+			// During each cache refresh cycle, it will check if any of these missed events gets populated.
```
Comment (azdagron) [inline pkg/server/endpoints/authorized_entryfetcher_registration_entries.go:54]:
```suggestion
			// During each cache refresh cycle, we will check if any of these missed events get populated.
```

File: pkg/server/endpoints/authorized_entryfetcher_registration_entries.go [hunk 7]
```diff
@@ -156,6 +158,10 @@ func (a *registrationEntries) updateCache(ctx context.Context) error {
 	return nil
 }
 
+// missedStartupEvents will check for any events come in with an ID less than the first event ID we receive.
+// For example if the first event ID we receive is 3, this function will check for any IDs less than that.
+// If event ID 2 comes in later on, due to a long running transaction, this function will update the cache
+// with the information from this event. This function will run the first sqlTransactionTimeout hours after startup.
```
Comment (azdagron) [inline pkg/server/endpoints/authorized_entryfetcher_registration_entries.go:164]: nit: sqlTransactionTimeout is not a unit of hours
```suggestion
// with the information from this event. This function will run until time equal to sqlTransactionTimeout has elapsed after startup.
```

**Review by azdagron — COMMENTED** [review]
Looks good. Just a few nits here and there.

**Review by azdagron — APPROVED** [review]

**Pull request merged**